lighthouse-core/lib/network-request.js

```diff
@@ -493,7 +493,23 @@ class NetworkRequest {
   static isSecureRequest(record) {
     return URL.isSecureScheme(record.parsedURL.scheme) ||
         URL.isSecureScheme(record.protocol) ||
-        URL.isLikeLocalhost(record.parsedURL.host);
+        URL.isLikeLocalhost(record.parsedURL.host) ||
+        NetworkRequest.isHstsRequest(record);
+  }
+
+  /**
+   * Returns whether the network request was an HSTS redirect request.
+   * @param {NetworkRequest} record
+   * @return {boolean}
+   */
+  static isHstsRequest(record) {
+    const destination = record.redirectDestination;
+    if (!destination) return false;
+
+    const reasonHeader = record.responseHeaders
+      .find(header => header.name === 'Non-Authoritative-Reason');
```

Comment:
ha, I wonder how this header ended up with this name

Reply:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.2.4

non-authorative because the info (the redirect) comes from a third party (hsts list in the browser), not the intended origin